### PR TITLE
Modify sentiment analysis book chapter 

### PR DESCRIPTION
**Pull request by sidgoyal78**

### CODE REVIEW

**Review by nickyfantasy — DISMISSED** [review]

File: 06.understand_sentiment/README.md [hunk 1]
```diff
@@ -98,18 +98,21 @@ We use [IMDB](http://ai.stanford.edu/%7Eamaas/data/sentiment/) dataset for senti
...
-# PaddlePaddle init
-paddle.init(use_gpu=False, trainer_count=1)
+import paddle
+import paddle.fluid as fluid
+from functools import partial
+import numpy as np
+
+CLASS_DIM = 2
+EMB_DIM = 128
+HID_DIM = 512
```
Comment (daming-lu) [inline 06.understand_sentiment/README.md:113]: I really think we should use `hidden` and correct the English misspellings sometime. For example `from paddle.v2.plot import Ploter`

Comment (daming-lu) [reply 06.understand_sentiment/README.md:113]: But it is not an urgent problem now.

File: 06.understand_sentiment/README.md [hunk 2]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
 
-    Parameter `input_dim` denotes the dictionary size, and `class_dim` is the number of categories. In `stacked_lstm_net`, the input to the network is defined in `paddle.layer.data`.
+```
+The above stacked bidirectional LSTM network extracts high-level features and maps them to a vector of the same size as the categories. `paddle.activation.Softmax` function or classifier is then used for calculating the probability of the sentence belonging to each category.
 
-1. Define Classifier
+To reiterate, we can either invoke `convolution_net` or `stacked_lstm_net`. In below steps, we will go with `convolution_net`.
 
-    The above stacked bidirectional LSTM network extracts high-level features and maps them to a vector of the same size as the categories. `paddle.activation.Softmax` function or classifier is then used for calculating the probability of the sentence belonging to each category.
+Next we define a `inference_program` that simply uses `convolution_net` to predict output with the input from `fluid.layer.data`.
```
Comment (daming-lu) [inline 06.understand_sentiment/README.md]: an `inference_program`

File: 06.understand_sentiment/README.md [hunk 3]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
-word_dict = paddle.dataset.imdb.word_dict()
-dict_dim = len(word_dict)
-class_dim = 2
+def train_program(word_dict):
+    prediction = inference_program(word_dict)
+    label = fluid.layers.data(name="label", shape=[1], dtype="int64")
+    cost = fluid.layers.cross_entropy(input=prediction, label=label)
+    avg_cost = fluid.layers.mean(cost)
+    accuracy = fluid.layers.accuracy(input=prediction, label=label)
+    return [avg_cost, accuracy]
```
Comment (daming-lu) [inline 06.understand_sentiment/README.md:212]: Here we should emphasize that the 1st returned parameter must be `cost`

File: 06.understand_sentiment/README.md [hunk 4]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
-    learning_rate=2e-3,
-    regularization=paddle.optimizer.L2Regularization(rate=8e-4),
-    model_average=paddle.optimizer.ModelAverage(average_window=0.5))
-
-# create trainer
-trainer = paddle.trainer.SGD(cost=cost,
-                                parameters=parameters,
-                                update_equation=adam_optimizer)
-```
+Notice for loading and reading IMDB data, it could take up to 1 minute. Please be patient.
```
Comment (daming-lu) [inline 06.understand_sentiment/README.md:235]: Mr. Minaj is a true gentleman 👍

Comment (jetfuel) [inline 06.understand_sentiment/README.md:235]: Does it always take 1 minute to load or just the first time?

File: 06.understand_sentiment/README.md [hunk 5]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
-            sys.stdout.write('.')
-            sys.stdout.flush()
-    if isinstance(event, paddle.event.EndPass):
-        with open('./params_pass_%d.tar' % event.pass_id, 'w') as f:
-            trainer.save_parameter_to_tar(f)
-
-        result = trainer.test(reader=test_reader, feeding=feeding)
-        print "\nTest with Pass %d, %s" % (event.pass_id, result.metrics)
+    if isinstance(event, fluid.EndStepEvent):
+        print("Step {0}, Epoch {1} Metrics {2}".format(
```
Comment (jetfuel) [inline 06.understand_sentiment/README.md:280]: This will print at every single step. It is too excessive. Can you change to every 100 steps or something like that?

File: 06.understand_sentiment/README.md [hunk 6]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
+                event.step, event.epoch, map(np.array, event.metrics)))
+
+        if event.step == 10:
```
Comment (jetfuel) [inline 06.understand_sentiment/README.md:283]: This is basically saying we will only train 10 steps. 
There is no need to call `trainer.stop()` since this is not a test.

File: 06.understand_sentiment/README.md [hunk 7]
```diff
@@ -118,212 +121,208 @@ As alluded to in section [Model Overview](#model-overview), here we provide the
...
 
+### Infer
+
+Now we can infer with inputs that we provide in `feed_order` during training.
+
+```python
+lod = [[3, 4, 2]]
+base_shape = [1]
+# The range of random integers is [low, high]
+tensor_words = fluid.create_random_int_lodtensor(
```
Comment (jetfuel) [inline 06.understand_sentiment/README.md]: Here is using a random vector. We should use a real example instead. (if possible.)

**Review by daming-lu — APPROVED** [review]
Tiny comments

**Comment by sidgoyal78** [comment]
Thanks @nickyfantasy for your efforts.

File: 06.understand_sentiment/README.md (hunk 4 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md:235]: it seems always

File: 06.understand_sentiment/README.md (hunk 6 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md:283]: this model will run like for 200steps and couple minutes with 1 epoch, I want to stop it fast so that user can see inference result in jupyter notebook

Comment (nickyfantasy) [reply 06.understand_sentiment/README.md:283]: In train.py, it will continue to run

File: 06.understand_sentiment/README.md (hunk 5 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md:280]: since we only run 10 steps in jupyter, i think it is fine to print all 10 steps. In train.py, I make it print every 10 steps

File: 06.understand_sentiment/README.md (hunk 7 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md]: Yes, I just did it with 3 sentences 'read the book forget the movie', 'this is a great movie', 'this is very bad'

File: 06.understand_sentiment/README.md (hunk 2 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md]: good grammar catch

File: 06.understand_sentiment/README.md (hunk 3 above)
Comment (nickyfantasy) [reply 06.understand_sentiment/README.md:212]: true

File: 06.understand_sentiment/README.md (hunk 6 above)
Comment (jetfuel) [reply 06.understand_sentiment/README.md:283]: How long does it take to run? If the training takes about 10 minutes or so, we can let the users know the time cost.

**Comment by daming-lu** [comment]
LGTM
